llvm/lib/Target/AMDGPU/SIInstrInfo.h

```diff
@@ -936,6 +936,14 @@ class SIInstrInfo final : public AMDGPUGenInstrInfo {
            Opcode == AMDGPU::S_BARRIER_SIGNAL_ISFIRST_IMM;
   }
 
+  bool isBarrierRelated(unsigned Opcode) const {
```

Comment:
Name could be better. Not sure what "barrier related" means, or why DS_GWS_BARRIER doesn't count as a barrier

Reply:
Change to `isBarrier` add DS_GWS parts.